### PR TITLE
Merging release-0.8 to master

### PR DESCRIPTION
**Pull request by hklai**
We are close to cutting a RC in release-0.8 for testing, so we are merging the commits in release-0.8 back to the master, to prepare for unlocking master.

### CODE REVIEW

**Comment by costinm** [comment]
Main problem seems to be lint - make depend.diff needs to be fixed or everyone will be blocked.

**Comment by hklai** [comment]
Discussed with @ldemailly and @costinm and we are merging this using an API version in the release-0.8 branch.

The latest API version in master is using a newer codegen (due to https://github.com/istio/api/pull/477) and the generated code is not compatible. It needs to be fixed separately.

/CC @geeknoid

**Comment by geeknoid** [comment]
I've got a pending PR to master which addresses the code gen stuff: https://github.com/istio/istio/pull/5623

I can merge this immediately to master, or immediately after release 0.8 is merged to master, or I could potentially merge it into the 0.8 branch. Please let me know what works best in the grand scheme of things.

**Comment by hklai** [comment]
TestTCPMixerFilter is failing in circle but passed in prow.

If it is a persistent issue we will deal with it later.

**Comment by hklai** [comment]
@geeknoid Please do not merge to release-0.8 if it is not needed in 0.8.x.

The codegen problem happened when I tried to use the latest API version, which I don't think #5623 is doing. Could you try updating the API version?

**Comment by ldemailly** [comment]
@geeknoid issues we found when updating api past your tooling change:
* gogo protobuf needed upgrade to 1.0.0 (for 0.5) and 1.1.0 (from 1.0.0) respectively to compile without getting InternalMessageInfo compile error on the new xxx_ generated code
* because a different generator is used for v1alpha1 and v1alpha3 networking, code in istioctl/convert needs to change to manually copy `Uri` and `Authority`

In short api tooling change should be preceded by trying to build istio/istio and immediately followed by an istio/istio PR to fix/adapt to the changes (or we should move istio/api inside istio/istio as was discussed)
